### PR TITLE
created new scheduler jobs

### PR DESCRIPTION
**Pull request by AdamDM**
wip, wrote tests and most of the implementation

finish tests and refactoring for new jobs

fixing for pr comments

wip: tinkering with tests

fix for PR comment

formatting

fix lists, httpclient, and timezone

fixing autowiring issues

adding jsse=false to dockerfile

fiddling with xml parsing

wip: start to sort out xml parsing issues

add example xml files for testing

make processing of xml document work for new calls

Revert "add example xml files for testing"

This reverts commit 8b758a89a578a4cd9f90a93e8d81272967e7a709.

### CODE REVIEW

**Comment by jamiesprax** [comment]
private final FiveBrandProcessor showProcessor;
private final FiveShowProcessor episodeProcessor;
EpisodeProcessingNodeFactory.create(brand, watchableProcessor);
private final FiveBrandWatchablesProcessor watchablesProcessor;

Something is really off with naming across all the processors here.

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import nu.xom.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.atlasapi.media.entity.Specialization.FILM;
+
+public class FiveBrandWatchablesProcessor extends FiveBrandProcessor{
+
+    private static final Logger log = LoggerFactory.getLogger(FiveBrandWatchablesProcessor.class);
+    private FiveWatchableProcessor watchableProcessor;
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: final?

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        watchableProcessor = FiveWatchableProcessor.create(
+                baseApiUrl,
+                httpClient,
+                channelMap,
+                locationPolicyIds
+        );
+    }
+
+    public static FiveBrandWatchablesProcessor create(
+            ContentWriter writer,
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: Builder.

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        EpisodeProcessingNodeFactory nodeFactory = EpisodeProcessingNodeFactory.create(
+                brand, watchableProcessor
+        );
+
+        try {
+            String responseBody = httpClient.get(getShowUri(id) + WATCHABLES_URL_SUFFIX).body();
+            new Builder(nodeFactory).build(new StringReader(responseBody));
+        } catch(Exception e) {
+            log.error("Exception parsing episodes for brand " + brand.getTitle(), e);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: Use formatting syntax, `log.error("... {}", brand.getTitle(), e)`

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        nodeFactory.items
+                .forEach(item -> write(brand, item));
+
+
+    }
+
+    private Brand extractBrand(Element element, String id) {
+        String uri = getShowUri(id);
+
+        Maybe<Identified> maybeBrand = contentResolver.findByCanonicalUris(
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: `Maybe`s are deprecated, there's a helper method to convert to either `Optional`.

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                ImmutableSet.of(uri)
+        )
+                .getFirstValue();
+
+        Brand brand = createBrand(element, id);
+
+        if (maybeBrand.hasValue()) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: Can be done via `Optional#ifPresent`

File: src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+    private Brand createBrand(Element element, String id) {
+        String uri = getShowUri(id);
+
+        Brand brand = new Brand(uri, getBrandCurie(id), Publisher.FIVE);
+        brand.setTitle(childValue(element, "title"));
+        brand.setDescription(getDescription(element).valueOrNull());
+        brand.setGenres(getGenres(element));
+
+        Maybe<Image> imageMaybe = getImage(element);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveBrandWatchablesProcessor.java]: Same, this is deprecated.

File: src/main/java/org/atlasapi/remotesite/five/FiveUpdater.java [hunk 7]
```diff
@@ -132,37 +134,146 @@ public void runTask() {
...
+                    "%s/shows?offset=%d&limit=%d",
+                    BASE_API_URL,
+                    startingPoint,
+                    LIMIT
+            );
+        }
     }
 
-    private void process(Elements elements) {
+    private void processShows(List<Element> elements, FiveBrandProcessor processor, String value) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/five/FiveUpdater.java]: `value` is a pretty bad name for an argument.

**Pull request closed without merging**